core/src/main/java/org/wildfly/glow/DeploymentScanner.java

```diff
@@ -193,6 +194,33 @@ private void scanAnnotations(DeploymentScanContext ctx) throws IOException {
                             }
                         }
                     }
+                    Map<String, List<AnnotationFieldValue>> fields = ctx.mapping.getAnnotationFieldValues().get(ai.name().toString());
+                    if (fields != null) {
+                        Layer foundLayer = null;
+                        for(Entry<String, List<AnnotationFieldValue>> f : fields.entrySet()) {
+                            String val = getAnnotationValue(ai, f.getKey());
+                            if (val != null) {
+                                List<AnnotationFieldValue> lstFields = f.getValue();
+                                for (AnnotationFieldValue fv : lstFields) {
+                                    if (Utils.isPattern(fv.getFieldValue())) {
+                                        Pattern p = Pattern.compile(fv.getFieldValue());
+                                        if (p.matcher(val).matches()) {
+                                            foundLayer = fv.getLayer();
+                                            LayerMapping.addRule(LayerMapping.RULE.ANNOTATION_VALUE, foundLayer, ai.name().toString() + "_" + f.getKey() + "=" + fv.getFieldValue());
+                                            ctx.layers.add(fv.getLayer());
+                                        }
+                                    } else {
+                                        if (val.equals(fv.getFieldValue())) {
+                                            foundLayer = fv.getLayer();
+                                            LayerMapping.addRule(LayerMapping.RULE.ANNOTATION_VALUE, foundLayer, ai.name().toString() + "_" + f.getKey() + "=" + fv.getFieldValue());
+                                            ctx.layers.add(fv.getLayer());
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                        handleDataSourceDefinitionAnnotations(ai, ctx, foundLayer);
```

Comment:
This will only be called with the last layer found in the above nested loops. If that's ok it could use a comment explaining why.

Reply:
Thank-you. Fixed.